drivers/wifi/nrf700x/CMakeLists.txt

```diff
@@ -129,14 +129,72 @@ zephyr_compile_definitions_ifdef(CONFIG_NRF700X_ON_QSPI
 )
 
 # RPU FW patch binaries based on the selected configuration
-zephyr_compile_definitions_ifdef(CONFIG_NRF700X_SYSTEM_MODE
-	-DCONFIG_NRF_WIFI_FW_BIN=${OS_AGNOSTIC_BASE}/fw_bins/default/nrf70.bin
+if(CONFIG_NRF700X_SYSTEM_MODE)
+	set (NRF70_PATCH ${OS_AGNOSTIC_BASE}/fw_bins/default/nrf70.bin)
+elseif (CONFIG_NRF700X_RADIO_TEST)
+	set (NRF70_PATCH ${OS_AGNOSTIC_BASE}/fw_bins/radio_test/nrf70.bin)
+elseif (CONFIG_NRF700X_SCAN_ONLY)
+	set (NRF70_PATCH ${OS_AGNOSTIC_BASE}/fw_bins/scan_only/nrf70.bin)
+else()
+	# Error
+	message(FATAL_ERROR "Unsupported nRF70 patch configuration")
+endif()
+
+zephyr_compile_definitions(
+	-DCONFIG_NRF_WIFI_FW_BIN=${NRF70_PATCH}
 )
 
-zephyr_compile_definitions_ifdef(CONFIG_NRF700X_RADIO_TEST
-	-DCONFIG_NRF_WIFI_FW_BIN=${OS_AGNOSTIC_BASE}/fw_bins/radio_test/nrf70.bin
+if (CONFIG_NRF_WIFI_PATCHES_EXT_FLASH_STORE)
+message(STATUS "nRF WiFi FW patch binary will be stored in external flash")
+
+if (CONFIG_BOARD_NRF5340DK_NRF5340_CPUAPP)
+set(NRF_QSPI_FLASH_BASE 0x10000000)
+elseif(CONFIG_BOARD_NRF52840DK_NRF52840)
+set(NRF_QSPI_FLASH_BASE 0x12000000)
+else()
+message(FATAL_ERROR "Unsupported board: ${BOARD}")
+endif()
+
+# Should be same as nrf70_fw_partition
+set(NRF70_PATCH_OFFSET 0x0)
+math(EXPR NRF70_PATCH_ADDR "${NRF_QSPI_FLASH_BASE} + ${NRF70_PATCH_OFFSET}")
+
+find_package(Python3)
+# detect virtualenv and set Pip args accordingly
+if(DEFINED ENV{VIRTUAL_ENV})
```

Comment:
I don't think this check is necessary. `intelhex` is a mandatory Zephyr dependency so nothing gets built without that.